### ISSUE TITLE
Homebridge not showing up in Home app

### ISSUE THREAD

**Issue by jamestenglish** [nfarina/homebridge#790]
I have homebridge running on a Raspberry Pi 2

```
$ node -v
v4.5.0
```

```
$ g++ -v
Using built-in specs.
COLLECT_GCC=g++
COLLECT_LTO_WRAPPER=/usr/lib/gcc/arm-linux-gnueabihf/4.9/lto-wrapper
Target: arm-linux-gnueabihf
Configured with: ../src/configure -v --with-pkgversion='Raspbian 4.9.2-10' --with-bugurl=file:///usr/share/doc/gcc-4.9/README.Bugs --enable-languages=c,c++,java,go,d,fortran,objc,obj-c++ --prefix=/usr --program-suffix=-4.9 --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --with-gxx-include-dir=/usr/include/c++/4.9 --libdir=/usr/lib --enable-nls --with-sysroot=/ --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --enable-gnu-unique-object --disable-libitm --disable-libquadmath --enable-plugin --with-system-zlib --disable-browser-plugin --enable-java-awt=gtk --enable-gtk-cairo --with-java-home=/usr/lib/jvm/java-1.5.0-gcj-4.9-armhf/jre --enable-java-home --with-jvm-root-dir=/usr/lib/jvm/java-1.5.0-gcj-4.9-armhf --with-jvm-jar-dir=/usr/lib/jvm-exports/java-1.5.0-gcj-4.9-armhf --with-arch-directory=arm --with-ecj-jar=/usr/share/java/eclipse-ecj.jar --enable-objc-gc --enable-multiarch --disable-sjlj-exceptions --with-arch=armv6 --with-fpu=vfp --with-float=hard --enable-checking=release --build=arm-linux-gnueabihf --host=arm-linux-gnueabihf --target=arm-linux-gnueabihf
Thread model: posix
gcc version 4.9.2 (Raspbian 4.9.2-10) 
```

```
$ DEBUG=* /opt/nodejs/bin/homebridge 
*** WARNING *** The program 'node' uses the Apple Bonjour compatibility layer of Avahi.
*** WARNING *** Please fix your application to use the native API of Avahi!
*** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=node>
*** WARNING *** The program 'node' called 'DNSServiceRegister()' which is not supported (or only supported partially) in the Apple Bonjour compatibility layer of Avahi.
*** WARNING *** Please fix your application to use the native API of Avahi!
*** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=node&f=DNSServiceRegister>
[9/22/2016, 5:24:59 PM] Loaded plugin: homebridge-http
[9/22/2016, 5:24:59 PM] Registering accessory 'homebridge-http.Http'
[9/22/2016, 5:24:59 PM] ---
[9/22/2016, 5:24:59 PM] Loaded config.json with 1 accessories and 0 platforms.
[9/22/2016, 5:24:59 PM] ---
[9/22/2016, 5:24:59 PM] Loading 1 accessories...
[9/22/2016, 5:24:59 PM] [Optoma Projector] Initializing Http accessory...
Scan this code with your HomeKit App on your iOS device to pair with Homebridge:

    ┌────────────┐     
    │ 031-45-155 │     
    └────────────┘     

  Accessory [Homebridge] Creating new AccessoryInfo for our HAP server +0ms
  Accessory [Homebridge] Creating new IdentifierCache +62ms
[9/22/2016, 5:25:00 PM] Homebridge is running on port 51826.
  EventedHTTPServer Server listening on port 51826 +275ms
  polling-to-event Emitting 'poll'. +0ms
  polling-to-event Comparing last polled parameters +18ms
  polling-to-event Last polled data and previous poll data are not equal. +10ms
  polling-to-event Emitting 'statuspoll'. +2ms
[9/22/2016, 5:25:01 PM] [Optoma Projector] Switch received power http://localhost/device/projector/status state is currently 0
[9/22/2016, 5:25:01 PM] [Optoma Projector] Setting power state to off
  polling-to-event Emitting 'poll'. +348ms
  polling-to-event Comparing last polled parameters +3ms
  polling-to-event Last polled data and previous poll data are equal. +7ms
....
```

I have the HTTP plugin installed and an accessories configured. Based on the above output it looks good

config.json

```
$ cat ~/.homebridge/config.json
{
    "bridge": {
        "name": "Homebridge",
        "username": "CC:22:3D:E3:CE:30",
        "port": 51826,
        "pin": "031-45-155"
    },

    "description": "This is an example configuration file with one fake accessory and one fake platform. You can use this as a template for creating your own configuration file containing devices you actually own.",

    "accessories": [
        {
                "accessory": "Http",
                "name": "Optoma Projector",
                "switchHandling": "realtime",
                "http_method": "GET",
                "on_url":      "http://localhost/device/projector/ON",
                "off_url":     "http://localhost/device/projector/OFF",
                "status_url":  "http://localhost/device/projector/status"
      } 
    ]

}
```

Webserver is running:

```
$ curl http://localhost/device/projector/status
0
```

Firewall is completely open: 

```
$ sudo iptables -L
Chain INPUT (policy ACCEPT)
target     prot opt source               destination         

Chain FORWARD (policy ACCEPT)
target     prot opt source               destination         

Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination 
```

I have delete the `~/.homebridge/persist` directory and restarted homebridge

When I go to the Home app on my iPhone and click **Add Accessory** nothing shows up.

I don't know what I am missing?

**Comment by KhaosT** [comment]
Can you use a bonjour browser on your iOS device and see if it can discovery the hap service homebridge advertised?

**Comment by cpives2008** [comment]
I'm running Homebridge on a RPi, not seeing anything when I look using a bonjour browser.

**Comment by toshibochan** [comment]
Try:
Sudo nano /home/pi/.homebridge/config.json

**Comment by RXM307** [comment]
On 23 September 2016 at 06:20, toshibochan <notifications@github.com> wrote:

> Try:
> Sudo nano /home/pi/.homebridge/config.json
>
>
> The other thing is Homekit Discovery seems to be very hit or miss, if I
connect my wifi via the switch on my router which is directly connecting my
pi I can see the home bridge but not via my normal wifi network which is
connected to the router via uplink to another switch, even though a bonjour
browser can see the mdns advertisements from the homebridge.

Cheers,

Anthony

**Comment by jamestenglish** [comment]
I don't see a HAP service listed in bonjour. There is a _homekit._tcp. but that is my apple tv. 

Is there a way to force/check the hap service to be advertised?

**Comment by jamestenglish** [comment]
This seems to have fixed the issue:
https://github.com/nfarina/homebridge/issues/607#issuecomment-236471712

**Comment by RXM307** [comment]
Yep that fixed my issue also, my router must have been dishing out ipv6
aswell as ipv4 whereas my normal wifi network was not.

Cheers,

Anthony

**Comment by RXM307** [comment]
Additionally I've finally fixed my issues, they were at the end of the day all bonjour issues, download the [Discovery - Bonjour Browser](https://itunes.apple.com/au/app/discovery-bonjour-browser/id305441017?mt=8) App and make sure you can see _hap._tcp. in the local domain, depending on your network setup you may need to enable multicast Forwarding or in my case my wireless controller need to know to snoop for "_hap._tcp.local."

**Comment by RXM307** [comment]
Additionally I also had bonjour issues #826

**Comment by jamiemcconnell** [comment]
I found that if I changed the `username` and restarted home bridge it showed right up.

**Comment by wexstorm** [comment]
Instead of changing the username and reading all devices to my home, I followed @jamestenglish Link and disabled IPV6. It is perfectly running now after rebooting!!!!
Thank you very much!

**Comment by sabs01** [comment]
I resolved this by uninstalling home bridge and reinstalling.  When there are plugins ie cameras which dont show up in homekit try uninstall these and reinstall them, then run homebridge all good.  They also showed in bonjour discovery browser under  the hap. tcp which was strange but by reinstalling them they must have reset their addresses or something.

**Comment by sithartha** [comment]
I had the same problem. 

RaspberryPi4 with homebridge on board.
I have a Xiaomi AC2100 router with OpenWRT on board.

At first I thought that the ipv6 was the problem, so I turned it off.
No luck.
Then I thought about Bonjour service, and enabled IGMP snooping on router.
No luck.
And at last I switched WIFI 5G to 2.4G. 

Profit.

Seems like Bonjour does not work at my net from eth to wifi 5G, only to wifi 2.4G
